### PR TITLE
feat(datatrak): RN-1313: My tasks section

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:

- Add new my tasks section to landing page

---

### Screenshots:
Screenshot 2024-07-16 at 12 38 00 PM

### CODE REVIEW

File: packages/datatrak-web/src/views/LandingPage/SurveySelectSection.tsx [hunk 1]
```diff
@@ -56,10 +72,11 @@ const ButtonWrapper = styled.div`
 `;
 
 const TextWrapper = styled.div`
-  margin: 0;
+  margin-bottom: 1rem;
```
Comment (alexd-bes) [inline packages/datatrak-web/src/views/LandingPage/SurveySelectSection.tsx:75]:
```suggestion
  margin-block-end: 1rem;
```

File: packages/datatrak-web/src/views/LandingPage/SurveySelectSection.tsx [hunk 2]
```diff
@@ -56,10 +72,11 @@ const ButtonWrapper = styled.div`
...
   display: flex;
   flex-direction: column;
   ${({ theme }) => theme.breakpoints.up('md')} {
+    margin-bottom: 0;
```
Comment (alexd-bes) [inline packages/datatrak-web/src/views/LandingPage/SurveySelectSection.tsx:79]:
```suggestion
    margin-block-end: 0;
```

File: packages/datatrak-web/src/views/LandingPage/TasksSection.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        <SectionHeading>My tasks</SectionHeading>
+        {hasTasks && (
+          <ViewMoreButton component={Link} to={ROUTES.TASKS}>
+            View more...
+          </ViewMoreButton>
+        )}
+      </FlexSpaceBetween>
+      <Paper>
+        {hasTasks && data?.tasks.map(task => <TaskTile key={task.id} task={task} />)}
+        {hasNoTasks && <NoTasksSection />}
```
Comment (alexd-bes) [inline packages/datatrak-web/src/views/LandingPage/TasksSection.tsx:73]: Does this section handle this part, as well? It does this if there are > 15 items and you hit the end of the list

Screenshot 2024-07-16 at 2 45 18 PM

**Review by alexd-bes — APPROVED** [review]
LGTM! I have a few finicky things and a couple of questions, but none of them are deal breakers, so pre-approving

File: packages/datatrak-web/src/features/Tasks/NoTasksSection.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  max-width: 100%;
+  width: 8rem;
+  margin: 0 auto 1rem;
+`;
+
+const Text = styled(Typography)`
+  text-align: center;
+  font-size: 0.93rem;
+  line-height: 1.5;
+  margin-bottom: 1rem;
```
[the remark this answers is not in the document]
Comment (tcaiger) [reply packages/datatrak-web/src/features/Tasks/NoTasksSection.tsx:31]: I'm still getting used to these properties :)

File: packages/datatrak-web/src/features/Tasks/TaskTile.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  display: flex;
+  align-items: center;
+
+  .MuiSvgIcon-root {
+    font-size: 1rem;
+    margin-inline-end: 0.2rem;
+  }
+`;
+
+// Todo: Update with actual comments count and set default to 0
```
[the remark this answers is not in the document]
Comment (tcaiger) [reply packages/datatrak-web/src/features/Tasks/TaskTile.tsx:75]: yes probably a good idea to hide this for now

File: packages/datatrak-web/src/views/LandingPage/TasksSection.tsx (hunk 3 above)
Comment (tcaiger) [reply packages/datatrak-web/src/views/LandingPage/TasksSection.tsx:73]: no I missed that. Thanks for pick it up

File: packages/datatrak-web/src/features/Tasks/NoTasksSection.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  padding: 0.25rem 1rem;
+`;
+export const NoTasksSection = () => (
+  <Container>
+    <Image />
+    <Text>
+      Congratulations, you have no tasks to complete! You can view all other tasks for your project
+      using the button below.
+    </Text>
+    <Button to={ROUTES.TASKS} component={Link}>
```
[the remark this answers is not in the document]
Comment (tcaiger) [reply packages/datatrak-web/src/features/Tasks/NoTasksSection.tsx:53]: hmm that is an interesting question. I think probably not at this point as that basically means defining the default filter setting to true

File: packages/datatrak-web/src/api/queries/useTasks.ts [hunk 7]
```diff
@@ -9,40 +9,35 @@ import { get } from '../api';
...
-              id: 'survey.project_id',
-              value: projectId,
-            },
-          ],
+          filters,
           sort: sortBy?.map(({ id, desc }) => `${id} ${desc ? 'DESC' : 'ASC'}`) ?? [],
         },
       }),
     {
-      enabled: !!projectId,
```
[the remark this answers is not in the document]
Comment (tcaiger) [reply packages/datatrak-web/src/api/queries/useTasks.ts:45]: thanks for checking this. I do not remember why I did this but I assume it was just me locally testing so I'm going to put it back

**Pull request merged**